pkg/csi_driver/node_test.go

```diff
@@ -199,6 +206,92 @@ func TestNodePublishVolume(t *testing.T) {
 	}
 }
 
+func TestWindowsNodePublishVolume(t *testing.T) {
+	defaultPerm := os.FileMode(0750) + os.ModeDir
+	defaultOsString := goOs
+
+	// Setup mount target path
+	base, err := ioutil.TempDir("", "node-publish-")
+	if err != nil {
+		t.Fatalf("failed to setup testdir: %v", err)
+	}
+	testTargetPath := filepath.Join(base, "mount")
+	if err = os.MkdirAll(testTargetPath, defaultPerm); err != nil {
+		t.Fatalf("failed to setup target path: %v", err)
+	}
+	defer os.RemoveAll(base)
+
+	goOs = "windows"
+
+	cases := []struct {
+		name          string
+		mounts        []mount.MountPoint // already existing mounts
+		req           *csi.NodePublishVolumeRequest
+		actions       []mount.FakeAction
+		expectedMount *mount.MountPoint
+		expectErr     bool
+		osString      string
+	}{
+		// TODO: enable this test when FakeMount supports Windows behavior
+		// {
+		// 	osString: "windows",
+		// 	name:     "windows target path does exist",
+		// 	req: &csi.NodePublishVolumeRequest{
+		// 		VolumeId:         testVolumeId,
+		// 		TargetPath:       testTargetPath,
+		// 		VolumeCapability: testVolumeCapability,
+		// 		VolumeAttributes: testVolumeAttributes,
+		// 		NodePublishSecrets: testWindowsSecrets
+		// 	},
+		// 	expectErr: true,
+		// },
+		{
+			osString: "windows",
+			name:     "windows target path doesn't exist",
+			req: &csi.NodePublishVolumeRequest{
+				VolumeId:           testVolumeId,
+				TargetPath:         testWindowsValidPath,
+				VolumeCapability:   testVolumeCapability,
+				VolumeAttributes:   testVolumeAttributes,
+				NodePublishSecrets: testWindowsSecrets,
+			},
+
+			actions:       []mount.FakeAction{{Action: mount.FakeActionMount}},
+			expectedMount: &mount.MountPoint{Device: testWindowsDevice, Path: testWindowsValidPath, Type: "cifs", Opts: []string{"foo", "bar"}},
+		},
+		{
+			osString: "windows",
+			name:     "windows no credentials",
```

Comment:
For 100% coverage you could split this into two tests: one where smbUser is not set (but password is), one where smbPassword is not set (but user is). Both should return errors.

Reply:
Done.